Lesson_10/Gift.js

```diff
@@ -0,0 +1,68 @@
+class Gift {
+    constructor() {
+      this.sweets = [];
+      this.weight = 0;
+      this.calories = 0;
+      this.minWeight = 0;
+      this.maxWeight = 0;
+      this.minCalories = 0;
+      this.maxCalories = 0;
+    }
+   setCalories() {
+      for (let i in this.sweets) {
+        this.calories += this.sweets[i].getCaloriesOfThisSweet();
+      }
+    }
+    getSweetsName() {
+      let sweetNames = ''
+      for (let i in this.sweets) {
+        sweetNames += this.sweets[i].name + ', ';
+      }
+      return sweetNames.slice(0, -2);
+    }
+    set sortedSweets(sortProperty) {
+      this.sweets = (sortProperty ? this.sweets.sort((a,b) => a[sortProperty] - b[sortProperty]) : this.sweets.sort((a,b) => a.name - b.name));
+    }
+    get sortedSweets() {
+      return this.sweets;
+    }
+    setWeightAndCalorieLimits() {
```

Comment:
if you set something you need to pass something, 
And this is little not a good method because this method do 2 thing, better create two methods one for wight one for calories